resources/tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/internal/resources/WorkspacePreferencesTest.java

```diff
@@ -243,48 +266,64 @@ public void testSetDescription() throws CoreException {
 		// try to make a change
 		description.setFileStateLongevity(100000);
 		// the original value should remain set
-		assertEquals("3.1", 90000, workspace.getDescription().getFileStateLongevity());
-		assertEquals("3.2", 90000, preferences.getLong(ResourcesPlugin.PREF_FILE_STATE_LONGEVITY));
-
-		setDefaultWorkspaceDescription();
+		assertThat(workspace.getDescription().getFileStateLongevity()).isEqualTo(90000);
+		assertThat(preferences.getLong(ResourcesPlugin.PREF_FILE_STATE_LONGEVITY)).isEqualTo(90000);
 	}
 
 	/**
 	 * Compares the values in a workspace description with the corresponding
 	 * properties in a preferences object.
 	 */
-	public void assertEquals(String message, IWorkspaceDescription description, Preferences expectedPreferences) throws ComparisonFailure {
-		assertEquals(message + " - 1", description.isAutoBuilding(), expectedPreferences.getBoolean(ResourcesPlugin.PREF_AUTO_BUILDING));
-		assertEquals(message + " - 2", description.getBuildOrder() == null, expectedPreferences.getBoolean(ResourcesPlugin.PREF_DEFAULT_BUILD_ORDER));
-		assertEquals(message + " - 3", WorkspacePreferences.convertStringArraytoString(description.getBuildOrder()), expectedPreferences.getString(ResourcesPlugin.PREF_BUILD_ORDER));
-		assertEquals(message + " - 4", description.isApplyFileStatePolicy(), expectedPreferences.getBoolean(ResourcesPlugin.PREF_APPLY_FILE_STATE_POLICY));
-		assertEquals(message + " - 5", description.getFileStateLongevity(), expectedPreferences.getLong(ResourcesPlugin.PREF_FILE_STATE_LONGEVITY));
-		assertEquals(message + " - 6", description.getMaxFileStates(), expectedPreferences.getInt(ResourcesPlugin.PREF_MAX_FILE_STATES));
-		assertEquals(message + " - 7", description.getMaxFileStateSize(), expectedPreferences.getLong(ResourcesPlugin.PREF_MAX_FILE_STATE_SIZE));
-		assertEquals(message + " - 8", description.getSnapshotInterval(), expectedPreferences.getLong(ResourcesPlugin.PREF_SNAPSHOT_INTERVAL));
-		assertEquals(message + " - 9", description.getMaxBuildIterations(), expectedPreferences.getLong(ResourcesPlugin.PREF_MAX_BUILD_ITERATIONS));
-		assertEquals(message + " -10", description.isKeepDerivedState(),
-				expectedPreferences.getBoolean(ResourcesPlugin.PREF_KEEP_DERIVED_STATE));
+	public void assertEquals(IWorkspaceDescription description, Preferences expectedPreferences)
```

Comment:
This could be made private in order to discourage inheritance and you could also flip the parameters so they follow the same logic as JUnit's `assertTrue` that is: first the **expected** value and then the **actual** value. Additionally, you could rename this method to something like `assertMatchesPreferences` or the likes so one can clearly see that this is a "custom" method and not a "JUnit" method (dealer's choice on the latter :-) ).

```suggestion
	private void assertEquals(Preferences expectedPreferences, IWorkspaceDescription description)
```

Reply:
Completely agreed. I've renamed the `assertEquals` methods in 46f98ac. I've also renamed the other `assertEquals` method, as this naming produces recurring problems due to name clashes with static imports of JUnit's `assertEquals` method.